src/common/utils/resolve_reference.py

```diff
@@ -245,8 +243,12 @@ class ResolvedReference:
     entity: dict | list
 
 
-def resolve_reference(address: Address, get_data_source: Callable) -> ResolvedReference:
-    """Resolve the reference into a document."""
+def resolve_address(address: Address, get_data_source: Callable) -> ResolvedReference:
```

Comment:
Should we also rename ResolvedReference to ResolvedAddress?

Reply:
yes, that sounds good